docs/release/trg-1/trg-1-08.md

```diff
@@ -0,0 +1,26 @@
+---
+title: TRG 1.08 - Interface documentation (APIs)
+sidebar_position: 1
+---
+
+| Status | Created     | Post-History                                          |
+|--------|-------------|-------------------------------------------------------|
+| Active | 22-Apr-2024 |                                                       |
+| Draft  | 21-Mar-2024 | Transferred from Release Acceptance Criteria into TRG |
+
+## Why
+
+Interface documentation is crucial to interact with components. It fundamentally enables others to consume your APIs.
+
+## Description
+
+- The interface documentation **must** be stored in the folder `/docs/api` as `openAPI.yaml`
+  - It **must** be in the OpenAPI standard
+  - It **must** contain all interfaces
+  - It **must** be up-to-date for the latest released component version
```

Comment:
I would vote for extending this to host the API documentation on Swagger Hub as well or we completely replace this part.

Side note: Tractus-X EDC is hosting it already on Swagger Hub: https://app.swaggerhub.com/apis/eclipse-tractusx-bot/tractusx-edc/

Reply:
@florianrusch-zf unfortunate swaggerhub limits the amount of projects to 3 or 5. Thats why we have to change this :|

Reply:
Yes swaggerhub unfortunately has this limit, but I'm also in favor of publishing them as artifact, as we're doing with jars and chart.
Github pages is a pretty good place to put them, we're already doing that in upstream EDC, e.g.:
https://eclipse-edc.github.io/Connector/openapi/management-api/
(specs at https://eclipse-edc.github.io/Connector/openapi/management-api/management-api.yaml)

different versions are also available by putting the version in the path:
https://eclipse-edc.github.io/Connector/openapi/management-api/3.0.1

Reply:
@ndr-brt we wanted to avoid having 30 api endpoints for people to go through. Also its effort for teams to build them and for us to verify them. With the openapi spec file, we can walk through them centrally, build a lightweight overview page and host all of them centrally.